### ISSUE TITLE
.datalad-git-backend.lock is left behind in the dataset root

### ISSUE THREAD

**Issue by mih** [datalad/datalad-metalad#92]
Not clear which of my explorations led to it, but:

```
% git st
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .datalad-git-backend.lock

nothing added to commit but untracked files present (use "git add" to track)
```

Other datalad pieces do something like this:

```
lockfile = self._repo_dot_git / 'config.dataladlock'
lockfile = Path(self.obtain('datalad.locations.cache')) / 'locks' / 'gitconfig.lck'
```

**Comment by bpoldrack** [comment]
> `lockfile = self._repo_dot_git / 'config.dataladlock'`

FTR: There's a `(Annex|Git)Repo.dot_git` attribute.

**Comment by christian-monch** [comment]
Now that metadata-store and dataset are considered the same, I implemented a version of the first proposed approach

**Comment by christian-monch** [comment]
Fixed in PR #123
